src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/WattTimeServerMocks.cs

```diff
@@ -0,0 +1,145 @@
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(200)
+                        .WithHeader("Content-Type", "application/json")
+                        .WithBody(json.ToString())
+                    );
+        }
+
+        /// <summary>
+        /// Setup forecast calls on mock server
+        /// </summary>
+        private static void SetupForecastMock(WireMockServer server)
+        {
+            var json = new JsonArray
+            {
+                new JsonObject
+                {
+                    ["generated_at"] = testDataPointOffset,
+                    ["forecast"] = new JsonArray
+                    {
+                        new JsonObject
+                        {
+                            ["ba"] = testBA,
+                            ["point_time"] = testDataPointOffset,
+                            ["value"] = 999.99,
+                            ["version"] = "1.0"
+                        }
+                    }
+                }
+            };
+
+            server
+                .Given(Request.Create().WithPath("/" + Paths.Forecast).UsingGet())
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(200)
+                        .WithHeader("Content-Type", "application/json")
+                        .WithBody(json.ToString())
+                    );
+        }
+        /// <summary>
+        /// Setup balancing authority calls on mock server
+        /// </summary>
+        private static void SetupBaMock(WireMockServer server)
+        {
+            var json = new JsonObject
+            {
+                ["id"] = "12345",
+                ["abbrev"] = testBA,
+                ["name"] = "Test Balancing Authority"
+            };
+            server
+                .Given(Request.Create().WithPath("/" + Paths.BalancingAuthorityFromLocation).UsingGet())
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(200)
+                        .WithHeader("Content-Type", "application/json")
+                        .WithBody(json.ToString())
+                    );
+        }
+
+        /// <summary>
+        /// Setup logins calls on mock server
+        /// </summary>
+        private static void SetupLoginMock(WireMockServer server)
+        {
+            server
+                .Given(Request.Create().WithPath("/" + Paths.Login).UsingGet())
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(200)
+                        .WithHeader("Content-Type", "application/json")
+                        .WithBody(JsonSerializer.Serialize(new LoginResult { Token = "myDefaultToken123" }))
+                    );
+        }
+
+        /// <summary>
+        /// Setup historical calls on mock server
+        /// </summary>
+        private static void SetupHistoricalMock(WireMockServer server)
```

Comment:
I don't think is implemented yet. So better to keep in sync with what is supported today imo.